pkg/generators/openapi.go

```diff
@@ -416,61 +409,24 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 	return nil
 }
 
-func isExtensionTag(tag string) bool {
-	isExtension := false
-	if strings.HasPrefix(tag, "x-kubernetes-") {
-		isExtension = true
-	}
-	return isExtension
-}
-
-// Returns sorted list of map keys. Needed for deterministic testing.
-func sortedMapKeys(m map[string]string) []string {
-	keys := make([]string, len(m))
-	i := 0
-	for k := range m {
-		keys[i] = k
-		i++
-	}
-	sort.Strings(keys)
-	return keys
-}
-
 func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
-	tagValues := getOpenAPITagValue(CommentLines)
-	type NameValue struct {
-		Name, Value string
-	}
-	extensions := []NameValue{}
-	// First, generate extensions from "+k8s:openapi-gen=x-kubernetes-*" annotations.
-	for _, val := range tagValues {
-		if isExtensionTag(val) {
-			parts := strings.SplitN(val, ":", 2)
-			if len(parts) != 2 {
-				return fmt.Errorf("invalid extension value: %v", val)
-			}
-			extensions = append(extensions, NameValue{parts[0], parts[1]})
-		}
-	}
-	// Next, generate extensions from "tags".
-	for _, tagKey := range sortedMapKeys(tagToExtension) {
-		tagValue, err := getSingleTagsValue(CommentLines, tagKey)
-		if err != nil {
-			return err
-		}
-		if len(tagValue) > 0 {
-			extensions = append(extensions, NameValue{tagToExtension[tagKey], tagValue})
-		}
-	}
-	// If there are generated extensions, write them out to schema.
-	// Example: "x-kubernetes-patch-strategy" : "merge"
+	extensions := parseExtensions(CommentLines)
 	if len(extensions) == 0 {
 		return nil
 	}
+	// If any extensions exist, then emit code to create them.
 	g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
 	for _, extension := range extensions {
-		g.Do("\"$.$\": ", extension.Name)
-		g.Do("\"$.$\",\n", extension.Value)
+		g.Do("\"$.$\": ", extension.name)
```

Comment:
it would be nicer if Extension has a method to generate its code. something like:

```golang
for _, extension := range extensions {
  extension.generate(g)
}
```

Reply:
This looks cleaner. But currently the Extension struct is a simple data object. It doesn't understand formatting or have any dependency on formatting objects. This type of function would add that dependency, and create code that is more tightly coupled.  But If we want to add this dependency, let's do it in a future PR.

Reply:
What dependency are you talking about? It only needs a type writer. Anything else? I think this way is much cleaner and I prefer if you do it here. If you feel strong about doing it in another PR please add the TODO and create an issue for it.

Reply:
@mbohlool Notice how simple, easy, and clean the unit tests are for Extension. Notice how the unit tests with the type writer are difficult to create, read, and understand. This is because the Extension struct has no formatting dependencies. Since the scope of this PR has already ballooned, we'll address this in a future PR.